diff --git a/templates/notify-widget-block.html.twig b/templates/notify-widget-block.html.twig
index 03ea9f5..62aa4e9 100644
--- a/templates/notify-widget-block.html.twig
+++ b/templates/notify-widget-block.html.twig
@@ -19,7 +19,7 @@
 <div id="notify_widget" {{ attributes.addClass(classes) }}>
   <a id="notify_toggle" href="#" title="Notifications">Notifications</a>
   {% if unread_count %}
-    <span class="unread">{{ unread_count|trans }}</span>
+    <span class="unread">{{ unread_count }}</span>
   {% endif %}
   <div id="notify_widget_popup">
     <h2>Notifications</h2>
