diff --git a/notify_widget.install b/notify_widget.install
index 53dffb2..af0c936 100644
--- a/notify_widget.install
+++ b/notify_widget.install
@@ -1,4 +1,4 @@
-<?php declare(strict_types=1);
+<?php
 
 /**
  * @file
diff --git a/notify_widget.module b/notify_widget.module
index 82502e4..347b06b 100644
--- a/notify_widget.module
+++ b/notify_widget.module
@@ -1,4 +1,4 @@
-<?php declare(strict_types = 1);
+<?php
 
 /**
  * @file
diff --git a/src/Access/NotificationPageAccessChecker.php b/src/Access/NotificationPageAccessChecker.php
index e2526ea..24359b5 100644
--- a/src/Access/NotificationPageAccessChecker.php
+++ b/src/Access/NotificationPageAccessChecker.php
@@ -1,17 +1,15 @@
-<?php declare(strict_types = 1);
+<?php
 
 namespace Drupal\notify_widget\Access;
 
 use Drupal\Core\Access\AccessResult;
-use Drupal\Core\Database\Connection;
 use Drupal\Core\Routing\Access\AccessInterface;
 use Drupal\Core\Session\AccountProxyInterface;
 use Drupal\user\Entity\User;
 
 /**
- * Checks user parameter matches the logged in user.
+ * Checks user parameter matches the logged-in user.
  *
- * Usage example:
  * @code
  * foo.example:
  *   path: '/example/{parameter}'
@@ -22,7 +20,7 @@
  *     _user_can_access_notifications: 'some value'
  * @endcode
  */
-final class NotificationPageAccessChecker implements AccessInterface {
+class NotificationPageAccessChecker implements AccessInterface {
 
   /**
    * Class constructor.
@@ -31,7 +29,7 @@ final class NotificationPageAccessChecker implements AccessInterface {
    *   The current user service.
    */
   public function __construct(
-    protected readonly AccountProxyInterface $currentUser
+    protected AccountProxyInterface $currentUser
   ) {}
 
   /**
diff --git a/src/Access/UserOwnsNotificationAccessChecker.php b/src/Access/UserOwnsNotificationAccessChecker.php
index ea2f27f..b24daa3 100644
--- a/src/Access/UserOwnsNotificationAccessChecker.php
+++ b/src/Access/UserOwnsNotificationAccessChecker.php
@@ -1,4 +1,4 @@
-<?php declare(strict_types = 1);
+<?php
 
 namespace Drupal\notify_widget\Access;
 
@@ -9,10 +9,10 @@
 use Drupal\user\Entity\User;
 
 /**
- * Checks if the user in the route parameter matches the owner of the
- * notification ID in the route parameter.
+ * Checks if the user in the route parameter matches.
+ *
+ * The owner of the notification ID in the route parameter.
  *
- * Usage example:
  * @code
  * foo.example:
  *   path: '/example/{parameter}'
@@ -23,7 +23,7 @@
  *     _user_owns_notification: 'some value'
  * @endcode
  */
-final class UserOwnsNotificationAccessChecker implements AccessInterface {
+class UserOwnsNotificationAccessChecker implements AccessInterface {
 
   /**
    * Constructs an UserOwnsNotificationAccessChecker object.
diff --git a/src/Controller/NotificationsController.php b/src/Controller/NotificationsController.php
index 25aa2fa..dba36b3 100644
--- a/src/Controller/NotificationsController.php
+++ b/src/Controller/NotificationsController.php
@@ -1,4 +1,4 @@
-<?php declare(strict_types = 1);
+<?php
 
 namespace Drupal\notify_widget\Controller;
 
@@ -12,13 +12,13 @@
 /**
  * Returns responses for Notify Widget routes.
  */
-final class NotificationsController extends ControllerBase {
+class NotificationsController extends ControllerBase {
 
   /**
    * The controller constructor.
    */
   public function __construct(
-    protected readonly NotifyWidgetApi $notifyWidgetApi,
+    protected NotifyWidgetApi $notifyWidgetApi,
   ) {}
 
   /**
@@ -135,7 +135,7 @@ public function __invoke(User $user = NULL): array {
    */
   public function markRead(int $notificationId) {
     $this->notifyWidgetApi->markAsRead($notificationId);
-    return new RedirectResponse(\Drupal::destination()->get());
+    return new RedirectResponse($this->getRedirectDestination()->get());
   }
 
   /**
@@ -146,7 +146,7 @@ public function markRead(int $notificationId) {
    */
   public function markUnread(int $notificationId) {
     $this->notifyWidgetApi->markAsUnread($notificationId);
-    return new RedirectResponse(\Drupal::destination()->get());
+    return new RedirectResponse($this->getRedirectDestination()->get());
   }
 
 }
diff --git a/src/Controller/NotifyWidgetController.php b/src/Controller/NotifyWidgetController.php
index ef91662..47c90b4 100644
--- a/src/Controller/NotifyWidgetController.php
+++ b/src/Controller/NotifyWidgetController.php
@@ -1,4 +1,6 @@
-<?php declare(strict_types = 1);
+<?php
+
+declare(strict_types=1);
 
 namespace Drupal\notify_widget\Controller;
 
@@ -13,20 +15,12 @@
  */
 class NotifyWidgetController extends ControllerBase {
 
-  /**
-   * The notify widget API service.
-   *
-   * @var \Drupal\notify_widget\NotifyWidgetApi
-   */
-  protected $notifyWidgetApi;
-
   /**
    * The controller constructor.
    */
   public function __construct(
-    NotifyWidgetApi $notifyWidgetApi
+    protected NotifyWidgetApi $notifyWidgetApi
   ) {
-    $this->notifyWidgetApi = $notifyWidgetApi;
   }
 
   /**
diff --git a/src/Form/DeleteNotificationConfirmForm.php b/src/Form/DeleteNotificationConfirmForm.php
index c836e5f..b6f2d46 100644
--- a/src/Form/DeleteNotificationConfirmForm.php
+++ b/src/Form/DeleteNotificationConfirmForm.php
@@ -1,4 +1,4 @@
-<?php declare(strict_types = 1);
+<?php
 
 namespace Drupal\notify_widget\Form;
 
@@ -10,9 +10,9 @@
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
- * @todo Add a description for the form.
+ * Delete notification confirmForm.
  */
-final class DeleteNotificationConfirmForm extends ConfirmFormBase {
+class DeleteNotificationConfirmForm extends ConfirmFormBase {
 
   /**
    * The notification ID from the route parameters.
@@ -28,7 +28,7 @@ final class DeleteNotificationConfirmForm extends ConfirmFormBase {
    *   The notify widget API service.
    */
   public function __construct(
-    protected readonly NotifyWidgetApi $notifyWidgetApi
+    protected NotifyWidgetApi $notifyWidgetApi
   ) {}
 
   /**
@@ -68,7 +68,8 @@ public function getCancelUrl(): Url {
    */
   public function buildForm(array $form, FormStateInterface $form_state, int|null $notificationId = NULL) {
     $this->notificationId = $notificationId;
-    return parent::buildForm($form, $form_state); // TODO: Change the autogenerated stub
+    // @todo Change the autogenerated stub.
+    return parent::buildForm($form, $form_state);
   }
 
   /**
diff --git a/src/Form/NotifyWidgetSendActionConfirmForm.php b/src/Form/NotifyWidgetSendActionConfirmForm.php
index df9aca0..1f480e5 100644
--- a/src/Form/NotifyWidgetSendActionConfirmForm.php
+++ b/src/Form/NotifyWidgetSendActionConfirmForm.php
@@ -1,4 +1,4 @@
-<?php declare(strict_types = 1);
+<?php
 
 namespace Drupal\notify_widget\Form;
 
@@ -15,58 +15,26 @@
 /**
  * Provides a system action for bulk notification sending.
  */
-final class NotifyWidgetSendActionConfirmForm extends ConfirmFormBase {
-
-  /**
-   * The temp store factory.
-   *
-   * @var \Drupal\Core\TempStore\PrivateTempStoreFactory
-   */
-  protected $tempStoreFactory;
-
-  /**
-   * The user storage.
-   *
-   * @var \Drupal\user\UserStorageInterface
-   */
-  protected $userStorage;
-
-  /**
-   * The entity type manager.
-   *
-   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
-   */
-  protected $entityTypeManager;
-
-  /**
-   * The notify widget API service.
-   *
-   * @var \Drupal\notify_widget\NotifyWidgetApi
-   */
-  protected $notifyWidgetApi;
+class NotifyWidgetSendActionConfirmForm extends ConfirmFormBase {
 
   /**
    * Constructs a new UserMultipleCancelConfirm.
    *
-   * @param \Drupal\Core\TempStore\PrivateTempStoreFactory $temp_store_factory
+   * @param \Drupal\Core\TempStore\PrivateTempStoreFactory $tempStoreFactory
    *   The temp store factory.
-   * @param \Drupal\user\UserStorageInterface $user_storage
+   * @param \Drupal\user\UserStorageInterface $userStorage
    *   The user storage.
-   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
    *   The entity type manager.
    * @param \Drupal\notify_widget\NotifyWidgetApi $notifyWidgetApi
    *   The notify widget API service.
    */
   public function __construct(
-    PrivateTempStoreFactory $temp_store_factory,
-    UserStorageInterface $user_storage,
-    EntityTypeManagerInterface $entity_type_manager,
-    NotifyWidgetApi $notifyWidgetApi
+    protected PrivateTempStoreFactory $tempStoreFactory,
+    protected UserStorageInterface $userStorage,
+    protected EntityTypeManagerInterface $entityTypeManager,
+    protected NotifyWidgetApi $notifyWidgetApi
   ) {
-    $this->tempStoreFactory = $temp_store_factory;
-    $this->userStorage = $user_storage;
-    $this->entityTypeManager = $entity_type_manager;
-    $this->notifyWidgetApi = $notifyWidgetApi;
   }
 
   /**
@@ -116,7 +84,6 @@ public function getConfirmText() {
     return $this->t('Send');
   }
 
-
   /**
    * {@inheritdoc}
    */
diff --git a/src/Form/NotifyWidgetSettingsForm.php b/src/Form/NotifyWidgetSettingsForm.php
index 10937e5..43710c5 100644
--- a/src/Form/NotifyWidgetSettingsForm.php
+++ b/src/Form/NotifyWidgetSettingsForm.php
@@ -1,4 +1,4 @@
-<?php declare(strict_types = 1);
+<?php
 
 namespace Drupal\notify_widget\Form;
 
@@ -8,7 +8,7 @@
 /**
  * Configure Notify Widget settings for this site.
  */
-final class NotifyWidgetSettingsForm extends ConfigFormBase {
+class NotifyWidgetSettingsForm extends ConfigFormBase {
 
   /**
    * {@inheritdoc}
@@ -46,7 +46,7 @@ public function buildForm(array $form, FormStateInterface $form_state): array {
       '#default_value' => $this->config('notify_widget.settings')->get('include_read') ?? TRUE,
     ];
 
-    $form['read_cutoff'] =[
+    $form['read_cutoff'] = [
       '#type' => 'select',
       '#title' => $this->t('Read Cut-Off'),
       '#description' => $this->t('Select a value from the list. Any read notifications that fall outside of the selected period will not show in the notifications pop-up.'),
diff --git a/src/NotifyWidgetApi.php b/src/NotifyWidgetApi.php
index 581cec9..933d809 100644
--- a/src/NotifyWidgetApi.php
+++ b/src/NotifyWidgetApi.php
@@ -1,4 +1,4 @@
-<?php declare(strict_types = 1);
+<?php
 
 namespace Drupal\notify_widget;
 
@@ -11,10 +11,10 @@
 /**
  * Provides an API for sending notifications to users.
  */
-final class NotifyWidgetApi {
+class NotifyWidgetApi {
 
   /**
-   * The notify widget module config.
+   * The notifies widget module config.
    *
    * @var \Drupal\Core\Config\Config
    */
@@ -24,11 +24,11 @@ final class NotifyWidgetApi {
    * Constructs a NotifyWidgetApi object.
    */
   public function __construct(
-    protected readonly Connection $database,
-    protected readonly ConfigFactoryInterface $configFactory,
-    protected readonly AccountProxyInterface $currentUser,
-    protected readonly CacheTagsInvalidatorInterface $cacheTagsInvalidator,
-    protected readonly TimeInterface $time
+    protected Connection $database,
+    protected ConfigFactoryInterface $configFactory,
+    protected AccountProxyInterface $currentUser,
+    protected CacheTagsInvalidatorInterface $cacheTagsInvalidator,
+    protected TimeInterface $time
   ) {
     $this->config = $this->configFactory->get('notify_widget.settings');
   }
@@ -124,7 +124,7 @@ public function send(
    * @param int $id
    *   The notification ID to set as read.
    */
-  public function markAsRead(int $id): void  {
+  public function markAsRead(int $id): void {
     $this->setNotificationStatus($id, 1);
   }
 
@@ -134,7 +134,7 @@ public function markAsRead(int $id): void  {
    * @param int $id
    *   The notification ID to set as unread.
    */
-  public function markAsUnread(int $id): void  {
+  public function markAsUnread(int $id): void {
     $this->setNotificationStatus($id, 0);
   }
 
diff --git a/src/NotifyWidgetLazyBuilders.php b/src/NotifyWidgetLazyBuilders.php
index 71c4b58..8e0539f 100644
--- a/src/NotifyWidgetLazyBuilders.php
+++ b/src/NotifyWidgetLazyBuilders.php
@@ -1,19 +1,20 @@
-<?php declare(strict_types = 1);
+<?php
 
 namespace Drupal\notify_widget;
 
-use DateTime;
 use Drupal\Component\Datetime\TimeInterface;
 use Drupal\Core\Config\ConfigFactoryInterface;
 use Drupal\Core\Database\Connection;
 use Drupal\Core\Security\TrustedCallbackInterface;
 use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
 use Drupal\user\UserData;
 
 /**
  * Defines a service for comment #lazy_builder callbacks.
  */
 class NotifyWidgetLazyBuilders implements TrustedCallbackInterface {
+  use StringTranslationTrait;
 
   /**
    * Constructs a new NotifyWidgetLazyBuilders object.
@@ -28,14 +29,16 @@ class NotifyWidgetLazyBuilders implements TrustedCallbackInterface {
    *   The config factory service.
    * @param \Drupal\Component\DateTime\TimeInterface $time
    *   The time service.
+   * @param \Drupal\user\UserData $userData
+   *   The user data service.
    */
   public function __construct(
-    protected readonly Connection $database,
-    protected readonly NotifyWidgetApi $notifyWidgetApi,
-    protected readonly AccountProxyInterface $currentUser,
-    protected readonly ConfigFactoryInterface $configFactory,
-    protected readonly TimeInterface $time,
-    protected readonly UserData $userData
+    protected Connection $database,
+    protected NotifyWidgetApi $notifyWidgetApi,
+    protected AccountProxyInterface $currentUser,
+    protected ConfigFactoryInterface $configFactory,
+    protected TimeInterface $time,
+    protected UserData $userData
   ) {}
 
   /**
@@ -56,7 +59,7 @@ public function renderNotifyWidgetBlock(): array {
     }
 
     // Get the notification details for the user.
-    $notifications = $this->notifyWidgetApi->getNotificationsForUser(0,0, $includeRead, $readCutoff);
+    $notifications = $this->notifyWidgetApi->getNotificationsForUser(0, 0, $includeRead, $readCutoff);
     $unreadCount = $this->notifyWidgetApi->getUnreadNotificationsCount();
 
     // If unread count is more than 20, set the count to 20+.
@@ -65,12 +68,12 @@ public function renderNotifyWidgetBlock(): array {
     }
 
     // Convert the timestamp for each notification to 'time ago'.
-    foreach ($notifications as $key => $notification) {
+    foreach ($notifications as $notification) {
       $notification->timestamp = $this->relativeTime($notification->timestamp);
     }
 
     // Get the current user's ID and the flag indicating whether they are
-    // new to the notifications system.
+    // new to the notification's system.
     $uid = $this->currentUser->id();
     $newToNotifications = empty($this->userData->get('notify_widget', $uid, 'new_to_notifications'))
       ? 'yes' :
@@ -114,45 +117,85 @@ public function renderNotifyWidgetBlock(): array {
    *   The timestamp to convert.
    *
    * @return string
+   *   Relative time.
    */
   private function relativeTime(int|string $ts): string {
-    if (!ctype_digit($ts))
+    if (!ctype_digit($ts)) {
       $ts = strtotime($ts);
+    }
 
     $diff = time() - $ts;
     if ($diff == 0) {
-      return 'now';
+      return $this->t('now');
     }
     elseif ($diff > 0) {
       $day_diff = floor($diff / 86400);
       if ($day_diff == 0) {
-        if ($diff < 60) return 'just now';
-        if ($diff < 120) return '1 minute ago';
-        if ($diff < 3600) return floor($diff / 60) . ' minutes ago';
-        if ($diff < 7200) return '1 hour ago';
-        if ($diff < 86400) return floor($diff / 3600) . ' hours ago';
+        if ($diff < 60) {
+          return $this->t('just now');
+        }
+        if ($diff < 120) {
+          return $this->t('1 minute ago');
+        }
+        if ($diff < 3600) {
+          return $this->t('@minutes minutes ago', ['@minutes' => floor($diff / 60)]);
+        }
+        if ($diff < 7200) {
+          return $this->t('1 hour ago');
+        }
+        if ($diff < 86400) {
+          return $this->t('@hours hours ago', ['@hours' => floor($diff / 3600)]);
+        }
+      }
+      if ($day_diff == 1) {
+        return $this->t('Yesterday');
+      }
+      if ($day_diff < 7) {
+        return $this->t('@day days ago', ['@day' => $day_diff]);
+      }
+      if ($day_diff < 14) {
+        return $this->t('1 week ago');
+      }
+      if ($day_diff < 31) {
+        return $this->t('@weeks weeks ago', ['@weeks' => ceil($day_diff / 7)]);
+      }
+      if ($day_diff < 60) {
+        return $this->t('last month');
       }
-      if ($day_diff == 1) return 'Yesterday';
-      if ($day_diff < 7) return $day_diff . ' days ago';
-      if ($day_diff < 14) return '1 week ago';
-      if ($day_diff < 31) return ceil($day_diff / 7) . ' weeks ago';
-      if ($day_diff < 60) return 'last month';
       return date('F Y', (int) $ts);
     }
     else {
       $diff = abs($diff);
       $day_diff = floor($diff / 86400);
       if ($day_diff == 0) {
-        if ($diff < 120) return 'in a minute';
-        if ($diff < 3600) return 'in ' . floor($diff / 60) . ' minutes';
-        if ($diff < 7200) return 'in an hour';
-        if ($diff < 86400) return 'in ' . floor($diff / 3600) . ' hours';
+        if ($diff < 120) {
+          return $this->t('in a minute');
+        }
+        if ($diff < 3600) {
+          return $this->t('in @minute minutes', ['@minute' => floor($diff / 60)]);
+        }
+        if ($diff < 7200) {
+          return $this->t('in an hour');
+        }
+        if ($diff < 86400) {
+          return $this->t('in @hours hours', ['@hours' => floor($diff / 3600)]);
+        }
+      }
+      if ($day_diff == 1) {
+        return $this->t('Tomorrow');
+      }
+      if ($day_diff < 4) {
+        return date('l', $ts);
+      }
+      if ($day_diff < 7 + (7 - date('w'))) {
+        return $this->t('next week');
+      }
+      if (ceil($day_diff / 7) < 4) {
+        return $this->t('in @weeks weeks', ['@weeks' => ceil($day_diff / 7)]);
+      }
+      if (date('n', $ts) == date('n') + 1) {
+        return $this->t('next month');
       }
-      if ($day_diff == 1) return 'Tomorrow';
-      if ($day_diff < 4) return date('l', $ts);
-      if ($day_diff < 7 + (7 - date('w'))) return 'next week';
-      if (ceil($day_diff / 7) < 4) return 'in ' . ceil($day_diff / 7) . ' weeks';
-      if (date('n', $ts) == date('n') + 1) return 'next month';
       return date('F Y', $ts);
     }
   }
diff --git a/src/Plugin/Action/NotifyWidgetSendAction.php b/src/Plugin/Action/NotifyWidgetSendAction.php
index 9f97dfe..607c5d8 100644
--- a/src/Plugin/Action/NotifyWidgetSendAction.php
+++ b/src/Plugin/Action/NotifyWidgetSendAction.php
@@ -1,4 +1,4 @@
-<?php declare(strict_types = 1);
+<?php
 
 namespace Drupal\notify_widget\Plugin\Action;
 
@@ -18,21 +18,7 @@
  *   confirm_form_route_name = "notify_widget.send_action_confirm",
  * )
  */
-final class NotifyWidgetSendAction extends ActionBase implements ContainerFactoryPluginInterface {
-
-  /**
-   * The tempstore factory.
-   *
-   * @var \Drupal\Core\TempStore\PrivateTempStoreFactory
-   */
-  protected $tempStoreFactory;
-
-  /**
-   * The current user.
-   *
-   * @var \Drupal\Core\Session\AccountInterface
-   */
-  protected $currentUser;
+class NotifyWidgetSendAction extends ActionBase implements ContainerFactoryPluginInterface {
 
   /**
    * Constructs a CancelUser object.
@@ -43,15 +29,12 @@ final class NotifyWidgetSendAction extends ActionBase implements ContainerFactor
    *   The plugin ID for the plugin instance.
    * @param mixed $plugin_definition
    *   The plugin implementation definition.
-   * @param \Drupal\Core\TempStore\PrivateTempStoreFactory $temp_store_factory
+   * @param \Drupal\Core\TempStore\PrivateTempStoreFactory $tempStoreFactory
    *   The tempstore factory.
-   * @param \Drupal\Core\Session\AccountInterface $current_user
+   * @param \Drupal\Core\Session\AccountInterface $currentUser
    *   Current user.
    */
-  public function __construct(array $configuration, $plugin_id, $plugin_definition, PrivateTempStoreFactory $temp_store_factory, AccountInterface $current_user) {
-    $this->currentUser = $current_user;
-    $this->tempStoreFactory = $temp_store_factory;
-
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, protected PrivateTempStoreFactory $tempStoreFactory, protected AccountInterface $currentUser) {
     parent::__construct($configuration, $plugin_id, $plugin_definition);
   }
 
diff --git a/templates/notify-widget-block.html.twig b/templates/notify-widget-block.html.twig
index afaf01c..03ea9f5 100644
--- a/templates/notify-widget-block.html.twig
+++ b/templates/notify-widget-block.html.twig
@@ -68,7 +68,7 @@
       {% else %}
         <div class="notification">
           <div class="text_wrapper">
-            <span class="text">You have no new notifications.</span>
+            <span class="text">{{ 'You have no new notifications.'|trans }}</span>
           </div>
         </div>
       {% endif %}
